managers/graph-operations-manager.js

```diff
@@ -8,6 +16,119 @@ class GraphOperationsManager {
         this.inputService = services.inputService;
     }
 
+    /**
+     * Retrieves a public or private assertion for a given UAL.
+     * @async
+     * @param {string} UAL - The Universal Asset Locator, representing asset or collection.
+     * @param {Object} [options={}] - Optional parameters for the asset get operation.
+     * @param {number} [options.state] - The state index of the asset. If omitted, the latest state will be used.
+     * @param {string} [options.contentType] - The type of content to retrieve, either "public" or "all" (default)
+     * @param {boolean} [options.validate] - Whether to validate the retrieved assertion.
+     * @param {string} [options.outputFormat] - The format of the retrieved assertion output, either "n-quads" or "json-ld".
+     * @returns {Object} - The result of the asset get operation.
+     */
+    async get(UAL, options = {}) {
+        const {
+            blockchain,
+            endpoint,
+            port,
+            maxNumberOfRetries,
+            frequency,
+            state,
+            contentType,
+            validate,
+            outputFormat,
+            authToken,
+            hashFunctionId,
+            paranetUAL,
+        } = this.inputService.getAssetGetArguments(options);
+
+        this.validationService.validateAssetGet(
+            UAL,
+            blockchain,
+            endpoint,
+            port,
+            maxNumberOfRetries,
+            frequency,
+            state,
+            contentType,
+            hashFunctionId,
+            validate,
+            outputFormat,
+            authToken,
+        );
+
+        const getOperationId = await this.nodeApiService.get(
+            endpoint,
+            port,
+            authToken,
+            UAL,
+            state,
+            contentType,
+            hashFunctionId,
+            paranetUAL,
+        );
+
+        const getOperationResult = await this.nodeApiService.getOperationResult(
+            endpoint,
+            port,
+            authToken,
+            OPERATIONS.GET,
+            maxNumberOfRetries,
+            frequency,
+            getOperationId,
+        );
+
+        const assertion = getOperationResult.data.assertion;
```

Comment:
we need to have consistent naming, in my PR i used dataset (assertion) and datasetRoot (assertionid)

I think final ones we agreed on are assertion and assertionRoot (not Id)